include/constants/battle.h

```diff
@@ -696,6 +696,7 @@ enum MoveTarget
 {
     TARGET_NONE,
     TARGET_SELECTED,
+    TARGET_SMART, // Like target select but can also smartly redirect to partner. Works only with strikeCount > 1 moves
```

Comment:
Can we have a test (or a static assert, but I think we need a test for this) to make sure no one uses ``TARGET_SMART`` with moves without ``strikeCount > 1``?